graphql-dgs/src/main/kotlin/com/netflix/graphql/dgs/internal/ReactiveDataFetcherResultProcessor.kt

```diff
@@ -52,4 +51,4 @@ class FluxDataFetcherResultProcessor : DataFetcherResultProcessor {
 }
 
 private fun reactorContextFrom(dfe: DgsDataFetchingEnvironment) =
-    DgsContext.from(dfe).reactorContext ?: Context.empty()
+    ReactiveDgsContext.from(dfe)?.reactorContext ?: Context.empty()
```

Comment:
Not sure if we should prefer throwing the `NullPointerException` here instead of all this null-handling. In theory with this change, there should always be a `ReactiveDgsContext` available in the environment. That relies on component configuration though, not the type checker... 🤔

Reply:
TLDR; Seems that this could sometimes be null when not depending on `graphql-dgs-spring-webflux-autoconfigure`.

The result processors are configured in `DgsAutoConfiguration` (used by a `DgsSchemaProvider`) whereas this new `ReactiveDgsContext` will only be constructed inside of the `DefaultDgsReactiveGraphQLContextBuilder` class made available by `DgsWebFluxAutoConfiguration`. Leaving the null checks for now is likely required.

You can use Mono/Flux in regular Spring Web MVC though. This overall PR would currently only work for Spring WebFlux apps; without modifying the existing `DefaultDgsQueryExecutor` to also be reactive-aware. I'm happy with that trade-off personally but should perhaps be documented.

Reply:
@gnoeley we have seen folks returning a _Mono_ or _Flux_ in Spring MVC, I don't think this approach will break those use-cases right?


Thanks for your changes, this PR LGTM